app/models/rule.rb

```diff
@@ -69,16 +70,20 @@ def canonical?
     (profiles & Profile.canonical).any?
   end
 
-  def self.from_openscap_parser(op_rule, benchmark_id: nil)
+  def self.from_openscap_parser(op_rule, benchmark_id: nil, group_id: nil)
     rule = find_or_initialize_by(ref_id: op_rule.id,
-                                 benchmark_id: benchmark_id)
+                                 benchmark_id: benchmark_id,
+                                 group_id: group_id)
```

Comment:
I'm not sure if we want to generate a new rule each time the group of an existing rule changes.

Reply:
You can check out usage of `create_with()` in [Rails guidelines](https://guides.rubyonrails.org/active_record_querying.html#find-or-create-by)